libs/wingc/src/type_check.rs

```diff
@@ -2584,6 +2596,49 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
+	/// Validate that the given type is a subtype (or same) as the expected type, disregarding whether
+	/// the object is mutable or immutable. If not, add an error to the diagnostics.
+	///
+	/// Returns the given type on success, otherwise returns the expected type.
+	fn validate_type_binary_operator(
```

Comment:
For now I think this behavior is specific to equality, maybe the function name can reflect that just to make sure it's not used elsewhere.